Tools/TestWebKitAPI/Tests/WebKitCocoa/AdditionalReadAccessAllowedURLs.mm

```diff
@@ -105,4 +113,83 @@
     TestWebKitAPI::Util::run(&done);
 }
 
+TEST(WebKit, NSAttributedStringWithTooManyReadOnlyPaths)
+{
+    __block bool done = false;
+
+    auto [readableDirectoryURL, unreadableDirectoryURL] = readableAndUnreadableDirectories();
+    NSURL *bundlePathURL = [NSURL fileURLWithPath:[[NSBundle mainBundle] bundlePath]];
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunguarded-availability-new"
+    NSMutableDictionary *options = [[NSMutableDictionary alloc] initWithObjectsAndKeys:@[ readableDirectoryURL, unreadableDirectoryURL, bundlePathURL ], WKReadAccessFileURLsOption, nil];
+#pragma clang diagnostic pop
+
+    bool exceptionRaised = false;
+    @try {
+        [NSAttributedString loadFromHTMLWithString:@"Hello World" options:options completionHandler:^(NSAttributedString *attributedString, NSDictionary<NSAttributedStringDocumentAttributeKey, id> *attributes, NSError *error) {
+            EXPECT_EQ([[attributedString attribute:NSFontAttributeName atIndex:0 effectiveRange:nil] pointSize], 12.0);
+
+            done = true;
+        }];
+        TestWebKitAPI::Util::run(&done);
+    } @catch (NSException *exception) {
+        EXPECT_WK_STREQ(NSInvalidArgumentException, exception.name);
+        exceptionRaised = true;
+    }
+    EXPECT_TRUE(exceptionRaised);
+}
+
+TEST(WebKit, NSAttributedStringWithReadOnlyPaths)
+{
+    __block bool done = false;
+
+    auto [readableDirectoryURL, unreadableDirectoryURL] = readableAndUnreadableDirectories();
+
+    NSURL *iconImagePath = [[NSBundle mainBundle] URLForResource:@"icon" withExtension:@"png" subdirectory:@"TestWebKitAPI.resources"];
+    NSURL *readableFileURL = [readableDirectoryURL URLByAppendingPathComponent:@"readable.png"];
+    NSError *error;
+    [NSFileManager.defaultManager copyItemAtURL:iconImagePath toURL:readableFileURL error:&error];
+    EXPECT_TRUE(!error || error.code == NSFileWriteFileExistsError);
+
+    NSURL *redImagePath = [[NSBundle mainBundle] URLForResource:@"large-red-square" withExtension:@"png" subdirectory:@"TestWebKitAPI.resources"];
+    NSURL *unreadableFileURL = [unreadableDirectoryURL URLByAppendingPathComponent:@"unreadable.png"];
+    [NSFileManager.defaultManager copyItemAtURL:redImagePath toURL:unreadableFileURL error:&error];
+    EXPECT_TRUE(!error || error.code == NSFileWriteFileExistsError);
+    
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunguarded-availability-new"
+    NSMutableDictionary *options = [[NSMutableDictionary alloc] initWithObjectsAndKeys:@[ readableDirectoryURL ], WKReadAccessFileURLsOption, nil];
```

Comment:
Ditto for leaking `options` here.

Reply:
Something, something, ARC. (Can't wait!)

Reply:
Fixed.